runtime/caml/io.h

```diff
@@ -106,6 +106,12 @@ CAMLextern intnat caml_really_getblock (struct channel *, char *, intnat);
 
 #define Channel(v) (*((struct channel **) (Data_custom_val(v))))
 
+/* The locking machinery */
+
+CAMLextern void (*caml_channel_mutex_lock) (struct channel *);
+CAMLextern void (*caml_channel_mutex_unlock) (struct channel *);
+CAMLextern void (*caml_channel_mutex_unlock_exn) (void);
```

Comment:
minor nit, but if we define this here then we should actually implement it in the `io.c` file.